### ISSUE TITLE
Fatal error in v8::ToLocalChecked:  Empty MaybeLocal when executing invalid code

### ISSUE THREAD

**Issue by jorrit** [audreyt/node-webworker-threads#73]
Node: 4.1.1
NWwT: 0.6.0

In my unit tests I deliberately execute an invalid sequence. This used to issue a regular warning, but since 0.6.0 the node process crashes. I have a core dump file if you need one, but it can be easily reproduced with the current master using the following code:

```javascript
const Threads = require('webworker-threads');
const pool = Threads.createPool(2);
pool.any.eval('something()');
```

**Comment by jorrit** [comment]
I've made a debug build with node-gyp and gdb tells me the error is at line 333 of WebWorkerThreads.cc:

```c++
if (!onError.HasCaught()) resultado= Nan::RunScript(script).ToLocalChecked();
```

**Comment by brodybits** [comment]
> This used to issue a regular warning, but since 0.6.0 the node process crashes.

On which version of this project did you see the "regular warning"?

I do see the crash at WebWorkerThreads.cc line 333 as well when running with Node 4.1.1. When running with Node 0.12.7 your test program seems to run with no output.

**Comment by brodybits** [comment]
Also does not crash on io.js 3.3.1 so it must be something introduced by Node 4.0. I would like to add that this project did not support Node 4.0 at all until very recently, as you probably already know.

**Comment by brodybits** [comment]
The following change seems to fix the crash on Node 4.0.0 and 4.1.1:

```diff
--- a/src/WebWorkerThreads.cc
+++ b/src/WebWorkerThreads.cc
@@ -330,7 +330,10 @@ static void eventLoop (typeThread* thread) {
 
             script= Nan::CompileScript(source).ToLocalChecked();
 
-            if (!onError.HasCaught()) resultado= Nan::RunScript(script).ToLocalChecked();
+            if (!onError.HasCaught()) {
+              MaybeLocal<Value> maybeResult = Nan::RunScript(script);
+              resultado = maybeResult.ToLocalChecked();
+            }
 
             if (job->typeEval.tiene_callBack) {
               job->typeEval.error= onError.HasCaught() ? 1 : 0;
```

Will issue a PR for review.

**Comment by brodybits** [comment]
I just issued PR #74 to fix this issue, waiting for feedback.

**Comment by brodybits** [comment]
Fix was approved and merged, closing.

**Comment by audreyt** [comment]
Thanks for the fix! Released as of 0.6.1. :rainbow:

**Comment by jorrit** [comment]
Thanks for the quick response! I don't recall what the old situation was, exactly, just that the callback argument to `pool.any.eval` received an error object. It appears to do that again, so many thanks!
